### ISSUE TITLE
alamut to show bam file

### ISSUE THREAD

**Issue by patrikgrenfeldt** [Clinical-Genomics/scout#1621]
From https://github.com/Clinical-Genomics/scout/issues/1460

**Comment by northwestwitch** [comment]
what about it?

**Comment by dnil** [comment]
I suppose pass URL for bam file stream with the Alamut button URL - or check again in the Alamut HTTP-control instructions how to point it to the bam file?

**Comment by northwestwitch** [comment]
Sounds quite easy, I'll try!

**Comment by dnil** [comment]
:+1: let me know if you need help testing on an alamut install later - or perhaps you have access to one already!

**Comment by northwestwitch** [comment]
> 👍 let me know if you need help testing on an alamut install later - or perhaps you have access to one already!

I knew that couldn't have been that easy!! :)

**Comment by northwestwitch** [comment]
There is already an Alamut link in cancer variants, the link points to some server that should be on localhost, port 10000. I don't have Alamut installed but maybe it works the same as for rare diseases?

**Comment by dnil** [comment]
Yes, which is good already, but the bam file will not tag along unless you tell it too..

**Comment by dnil** [comment]
Probably

```
request=BAM%3C"+bamFile
```
based on https://www.interactive-biosoftware.com/doc/alamut-visual/2.8/programmatic-access.html

**Comment by dnil** [comment]
Currently the Alamut Visual Plus API accepts local bam and cram to the open command, and there is mention that only bam files will work with a remote (http) url. It also does not seem to be able to do ranged requests, rather uses the index file directly to retrieve portions after downloading the whole file. Hence, blocked until further noticed. Maybe we should just  close this and reopen when the API changes?

**Comment by molucorner** [comment]
Is it time to repoen this issue again? From what I've seen when I've tried it, it seems to only go for the reads in the region that's open in Alamut.

**Comment by dnil** [comment]
Hi! Have they updated the API with any of the latest updates? I haven't checked for a few versions, but then one had to have the whole alignment downloaded to a filesystem accessible by Alamut for it to work? They did not support standard ga4gh alignment streaming range requests e.g.

**Comment by dnil** [comment]
Ah, indeed, the documentation seems to have updated a bit at least! Maybe worth a shot!

![Image](https://github.com/user-attachments/assets/980c9b4a-a5c1-41b0-8178-05548ad5d528)

**Comment by dnil** [comment]
Super! The files not really working in Alamut was the big issue before. There are a few standard ways off allowing access outside the session. They usually involve the creation of a more or less specific and temporally or destination constrained token for access to what is otherwise a privileged endpoint. Keeping a list of more or less random tokens issued is usually required unless very elaborate crypto is used. Perhaps first take a look among the Flask extensions to see if there is a good library to handle this, otherwise I'm sure we can cook a good enough one.
